rpc/connection.go

```diff
@@ -503,12 +503,13 @@ func (c *Connection) DoCommand(ctx context.Context, name string,
 }
 
 // Blocks until a connnection is ready for use or the context is canceled.
-func (c *Connection) waitForConnection(ctx context.Context) error {
+func (c *Connection) waitForConnection(
+	ctx context.Context, forceReconnect bool) error {
 	reconnectChan, disconnectStatus, reconnectErrPtr, wait :=
 		func() (chan struct{}, DisconnectStatus, *error, bool) {
 			c.mutex.Lock()
 			defer c.mutex.Unlock()
-			if c.isConnectedLocked() {
+			if !forceReconnect && c.isConnectedLocked() {
```

Comment:
I think you need to set `c.reconnectChan` to nil, otherwise this happens:

https://github.com/keybase/go-framed-msgpack-rpc/blob/master/rpc/connection.go#L575-L577

Reply:
Hmm I think it should be `nil` as long as no other attempt to reconnect is ongoing: https://github.com/keybase/go-framed-msgpack-rpc/blob/3b7dbb848bdcf9c6c463b034d4195aab3c80e8d5/rpc/connection.go#L634

Reply:
Cool, I stopped the inception 1 level too early.